Mr/gradle classes

marschwar opened the pull request:
I fixed the tests and removed the dependencies to `CheckstyleReports` and `AbstractCodeQualityPlugin`.

I will write more tests in the days to come.

Make sure to run ` ./gradlew clean publishToMavenLocal` to create the `1.0.0-GRADLE` version of the cli

3flex committed:
CI: Improve Windows testing

Run equivalent commands in Windows CI environment as *nix CI

Mauin committed:
Merge remote-tracking branch 'upstream/master'

3flex committed:
Fix UnsafeCast rule violation

schalkms committed:
Fixed MayBeConst false negative when concatenating strings #900

schalkms committed:
Added support for parenthesized expressions for MayBeConst

schalkms committed:
Fixed some IntelliJ inspection warnings in test cases

Commit:
TooManyFunctions : Add option to ignore private functions

Commit:
Update resources

Commit:
Refactor code

Commit:
Coding style

Commit:
Remove wrong import

arturbosch committed:
Generate javadocJar for maven central publishing - #345

arturbosch committed:
Prepare RC7-2

arturbosch committed:
Update gradle plugin versions

arturbosch committed:
Use latest detekt version

arturbosch committed:
Respect string interpolation - Closes #808

schalkms committed:
Added tests for MayBeConstSpec

schalkms committed:
Added MayBeConstSpec to support const in nested companion objects

scottkennedy committed:
Add a space between two words

"isunused" -> "is unused"

arturbosch committed:
Provide signing config for bintray - #345

arturbosch committed:
Do not crash on rule exceptions - #793 #944

arturbosch committed:
Suppress dokka warnings for undocumented classes

arturbosch committed:
Print detekt- and java versions with os name on rule exceptions

arturbosch committed:
Close manifest streams

arturbosch committed:
Suppress stacktrace for build failure exception when maxIssues met

arturbosch committed:
Prepare RC7-3 release

epeee committed:
make gradlew executable

epeee committed:
Use native junitPlatform from gradle 4.6 in detekt-gradle-plugin project

epeee committed:
Use native junitPlatform from gradle 4.6

schalkms committed:
Updated ExpressionBodySyntax to allow for exclusion on line wrapping

reference #950

schalkms committed:
Updated ExpressionBodySyntax documentation

schalkms committed:
Added ExpressionBodySyntax config 'includeLineWrapping'

vanniktech committed:
WildcardImport: Remove one force unwrap and reuse it from let function

schalkms committed:
Updated kotlin compiler to version 1.2.50

epeee committed:
Migrate detekt-formatting/build.gradle to Kotlin DSL

vanniktech committed:
Allow "All" in @Suppress statements

schalkms committed:
Fixed #960 - private properties naming report

Commit:
Fix SerialVersionUIDInSerializableClass when value's less then zero #964

arturbosch committed:
Support suppression of single instances of MaxLineLength violations - Closes #715

arturbosch committed:
Change ruleSets to plugins in gradle dsl extension docs

marianosimone committed:
UnusedPrivateMember: detect top level declarations

marianosimone committed:
UnusedPrivateMember: Detect unused properties and params in constructors

arturbosch committed:
Use detekt 7-3 in self analysis

arturbosch committed:
Update changelog

marianosimone committed:
UnusedPrivateMember: do not report unused parameters in abstract/open functions

Fixes #972

marianosimone committed:
Fix ExpressionBodySyntax when multiline expression

Check the body to see if it's multiline, instead of the
return expression, which might be multiline, but with
none of its children having a \n

Fixes #976

schalkms committed:
Added some rules to failfast.yml

marianosimone committed:
New rule: VarCouldBeVal

schalkms committed:
Added rules to failfast.yml

* Removed unused imports

schalkms committed:
Ignore unused operator function parameters

The UnusedPrivateMember rule is compliant with IntelliJs rule for
unused operator function parameters.

schalkms committed:
Cleanup

schalkms committed:
Removed unused variables in build script

schalkms committed:
Fixed unused fun detection in main

A function which is only called from the main function was detected as
unused.

schalkms committed:
Cleanup

arturbosch committed:
Mention --run-rule option and get_analysis_projects script

schalkms committed:
Exclude companion obj from NestedClassVisibility

schalkms committed:
Update kotlin version to 1.2.51

schalkms committed:
Refactor UndocumentPublicFunction rule

schalkms committed:
Added rule to failfast.yml

schalkms committed:
Added rule to failfast.yml

schalkms committed:
Updated StringLiteralDuplication message

schalkms committed:
Merged failfast.yml

marianosimone committed:
Refactor UnusedImports rule

1. Make tests a bit more robust (by actually verifying what the
smells are)
2. Implement the rule with an internal visitor, so there's no state
in the rule itself (which needed to be reset for each file, preventing
it for being a candidate for parallel excecution)
3. Only look at KtReferenceExpressions that are not imports, package
directives nor have children. For example:

foo(a, b)

actually gets visited for `foo(a(b), c)`, `foo`, `a(b)`, `a`, `b`, and
`c`, but we only care about `foo`, `a`, `b`, and `c`

arturbosch committed:
Update and activate ktlint in self-analysis - #925

arturbosch committed:
Format codebase with ktlint - #925

arturbosch committed:
Allow autoCorrect only if ruleSet also allows it - #925

arturbosch committed:
Drop support for RuleTest class

arturbosch committed:
Use new loadRuleSet function

arturbosch committed:
Fix SafeCast issue

arturbosch committed:
Document formatting rule set - Closes #925

marianosimone committed:
UnusedImports: Add detection of imports in same package

Imports in the same package are unnecessary, and can be safely removed
Fixes #985

duck8823 committed:
Fix to work on multi module in pom.xml

arturbosch committed:
Fix magic number report for named constructor calls - Fixes #992

arturbosch committed:
Fix false positive lambda in constructor call - Fixes #990

arturbosch committed:
Refactor ProfileStorage to be no singleton - Fixes #980

Commit:
fixes version problem

Commit:
fixes tests
 - plugin should not set default config file
 - properly trim multi line xml
 - ignore custom task test for now

marianosimone committed:
UnusedImports: Add test for inner classes in same package

This makes sure that references to inner classes from outer ones
can be imported without being detected as a smell

arturbosch committed:
Simplify test case

Commit:
removes AbstractCodeQualityPlugin inheritance

schalkms committed:
Flag empty nested functions - fixes #998

schalkms committed:
Flag empty nested catch clause

schalkms committed:
Flag empty nested expressions

egor-n committed:
Add a failing unit test

egor-n committed:
Add prefix wildcard to SplitPattern

Commit:
adds reporting functionality and tests

scottkennedy committed:
Add a missing space

Commit:
clean up extension and task properties

Commit:
create custom reports extension to allow configuration in detekt block.

So instead of
detekt {
  toolVersion = "1.0.0-GRADLE"
}

tasks.withType<io.gitlab.arturbosch.detekt.Detekt> {
  reports {
    xml.isEnabled = true
    xml.destination = file("build/reports/detekt.xml")
    html.isEnabled = true
    html.destination = file("build/reports/detekt.html")
  }
}

it is possible to write:

detekt {
  toolVersion = "1.0.0-GRADLE"
  reports {
    xml.enabled = true
    xml.destination = file("build/reports/detekt.xml")
    html.enabled = true
    html.destination = file("build/reports/detekt.html")
  }
}

scottkennedy committed:
Add another missing space

Commit:
Adds support for multi module project

marschwar commented:
@Mauin As far as I can tell, the check task should now work with a root project and a multi module project. If not specified in the submodule, the properties are drawn from the parents.
Should we try to rebase onto the master?
What else needs to be done?

schalkms committed:
Add aliases to rule documentation

* Aliases are read with reflection
* Reference #979

schalkms committed:
Regenerate aliases rule documentation

schalkms committed:
Robustify documentation generator for issue declaration

The generator now throws an exception if the issue does not conform to
the style declaration.

schalkms committed:
Moved documentation generator exception to own package

schalkms committed:
Update README

Commit:
Detekt task does not longer extend SourceTask.
This led to each individual source file to be added to the --input parameter. For larger projects this does not seem to be a good idea. If we decided to execute the DetektFacade directly, this could become an option.

Commit:
avoid creating html and xml output on any other task than check

Commit:
cleanup some code

arturbosch committed:
Update gradle-build dependencies

arturbosch committed:
Document dokka exclusion from build to save time for contributors

arturbosch committed:
Fix gradle-plugin's group to publish it again under the same bintray repository

arturbosch committed:
Fix detekt issues

arturbosch committed:
Configure bintray uploading for the gradle-plugin

arturbosch committed:
Update detekt version to RC8

Commit:
removes use of conventions mapping alltogether.

see https://guides.gradle.org/implementing-gradle-plugins/
"Please avoid using the Convention concept when writing new plugins. The long term plan is to migrate all Gradle core plugins to use extensions and remove the Convention concept altogether."

Commit:
Extract cli invocation arguments to classes

Commit:
add test for create baseline task

Commit:
create test for generate config task

arturbosch committed:
Update used detekt version

Commit:
removes reference to detekt task from idea tasks

Commit:
creates detekt task only if needed

Commit:
removes unused constants

Commit:
cleans up code

Commit:
finally gets rid of java.lang.Boolean workaround

Commit:
removes usages of Property<> and Provider<>

Mauin commented on detekt-gradle-plugin/src/main/kotlin/io/gitlab/arturbosch/detekt/Detekt.kt:
```diff
@@ -1,119 +1,88 @@
 
 import groovy.lang.Closure
-import io.gitlab.arturbosch.detekt.invoke.DetektInvoker
+import io.gitlab.arturbosch.detekt.invoke.*
```
detekt won't like the star imports.

Mauin commented on detekt-gradle-plugin/src/main/kotlin/io/gitlab/arturbosch/detekt/Detekt.kt:
```diff
@@ -1,119 +1,88 @@
-		}
-		return map
-	}
+open class Detekt : DefaultTask(), Reporting<DetektReports> {
```
`SourceTask` indicates that the task will perform actions on sources, which doesn't sound too wrong. Why the change to `DefaultTask`? https://docs.gradle.org/current/javadoc/org/gradle/api/tasks/SourceTask.html

marschwar replied:
IMHO extending SourceTask does not have any benefits unless we decide to actually use the includes/excludes mechanism. But in order to do so we should rework the way we pass source files to detekt as it right now does the filtering internally.

Mauin commented on detekt-gradle-plugin/src/main/kotlin/io/gitlab/arturbosch/detekt/extensions/DetektExtension.kt:
```diff
@@ -1,75 +1,59 @@
+	val defaultReportsDir = project.layout.buildDirectory.get()
+			.dir(ReportingExtension.DEFAULT_REPORTS_DIR_NAME)
+			.dir("detekt").asFile
+	val defaultSourceDirectories = project.files("src/main/java", "src/main/kotlin")
```
Is it possible to dynamically get the source set paths with `project.sourceSets`?

Commit:
Merge branch 'master' of github.com:arturbosch/detekt

Commit:
Merge branch 'mr/gradle_classes' of github.com:marschwar/detekt into mr/gradle_classes

Commit:
change detekt configFile to config

AndreasVolkmann committed:
Replace // with # in yaml code

`//` is not a valid comment in yaml. When copying this code, it will not parse and you have to manually replace with `#`.

Commit:
Merge branch 'master' into mr/gradle_classes

Commit:
use custom detekt version for build

egor-n committed:
Change description of TopLevelPropertyNaming rule

Commit:
fixes reports and idea extension configuration for kotlin dsl

Commit:
reformats code according to idea-code-style.xml

Commit:
uses constant LifecycleBasePlugin.VERIFICATION_GROUP

Commit:
Adds reports to default detekt config

Commit:
Removes code smells

Commit:
Updates documentation to new gradle plugin

marschwar commented:
@Mauin do you have any idea how to solve the chicken-and-egg-problem that the we can only build the latest gradle plugin with the latest gradle plugin? Especially now that the version is `1.0.0-GRADLE`. Locally I did:
```
./gradlew clean publishToMavenLocal -x dokka -x detekt
./gradlew build -PwarningsAsErrors=true
```

Commit:
Documents usage of input property in gradle extension

Commit:
Moves constants that describe the default configuration into the DetektExtension

Commit:
Fixes test that creates a custom detekt task

Commit:
Removes application of ReportingBasePlugin as ReportingExtension was never read

Commit:
Extends test with custom task to report destinations

Commit:
Removes the gradle reporting container to avoid usage of internal API

- changes example for kotlin dsl to create a custom task
- tests creation of custom tasks using groovy and kotlin dsl

Commit:
Merge remote-tracking branch 'upstream/master' into mr/gradle_classes
